src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -727,7 +595,24 @@ public void scheduleAtFixedRate(Runnable task, long initialDelay, long period,
       priority = defaultPriority;
     }
 
-    addToScheduleQueue(new RecurringRateTaskWrapper(task, priority, initialDelay, period));
+    addToScheduleQueue(priority, new RecurringRateTaskWrapper(task, getQueueManager(priority), initialDelay, period));
+  }
+  
+  /**
+   * Simple class to return the appropriate {@link QueueManager} for a given priority.
+   * 
+   * @param priority Priority to check against for finding the queue
+   * @return Task queue which makes sense for the given priority 
+   */
+  protected QueueManager getQueueManager(TaskPriority priority) {
```

Comment:
I removed several switch statements and instead just have this one to return the appropriate QueueManager